### PR TITLE
Updates the transaction API page and cookbook.

### PR DESCRIPTION
**Pull request by joeldudleyr3**

### CODE REVIEW

File: docs/source/api-transactions.rst [hunk 1]
```diff
@@ -11,12 +11,11 @@ API: Transactions
 
 Transaction workflow
 --------------------
-There are four states the transaction can occupy:
+At any time, a transaction can occupy one of three states:
 
-* ``TransactionBuilder``, a builder for a transaction in construction
-* ``WireTransaction``, an immutable transaction
```
Comment (Clintonio) [inline docs/source/api-transactions.rst]: I see WireTransaction in the repo still, how come it is removed?

File: docs/source/api-transactions.rst [hunk 2]
```diff
@@ -357,31 +356,32 @@ transaction's signatures.
...
-       :end-before: DOCEND 31
+       :start-after: DOCSTART 33
+       :end-before: DOCEND 33
        :dedent: 12
 
-However, this still isn't enough. The ``WireTransaction`` holds its inputs as ``StateRef`` instances, and its
-attachments as hashes. These do not provide enough information to properly validate the transaction's contents. To
-resolve these into actual ``ContractState`` and ``Attachment`` instances, we need to use the ``ServiceHub`` to convert
-the ``WireTransaction`` into a ``LedgerTransaction``:
+We will generally also want to conduct some additional validation of the transaction, beyond what is provided for in
```
Comment (Clintonio) [inline docs/source/api-transactions.rst]: This line could be simplified. It's hard to follow.

File: docs/source/example-code/src/main/java/net/corda/docs/FlowCookbookJava.java [hunk 3]
```diff
@@ -425,41 +429,43 @@ protected void verifyDataRequest(@NotNull FetchDataFlow.Request.Data dataRequest
...
+                // into a ``LedgerTransaction``. This will use our ServiceHub
+                // to resolve the transaction's inputs and attachments into
+                // actual objects, rather than just references.
+                // DOCSTART 32
+                LedgerTransaction ledgerTx = twiceSignedTx.toLedgerTransaction(getServiceHub());
+                // DOCEND 32
+
+                // We can now perform our additional verification.
+                // DOCSTART 34
+                DummyState outputState = (DummyState) ledgerTx.getOutputs().get(0).getData();
```
Comment (Clintonio) [inline docs/source/example-code/src/main/java/net/corda/docs/FlowCookbookJava.java]: Is there a utility function to simplify this call?

File: docs/source/api-transactions.rst (hunk 1 above)
Comment (joeldudleyr3) [reply docs/source/api-transactions.rst]: Discussed.

File: docs/source/api-transactions.rst (hunk 2 above)
Comment (joeldudleyr3) [reply docs/source/api-transactions.rst]: Addressed.

File: docs/source/example-code/src/main/java/net/corda/docs/FlowCookbookJava.java (hunk 3 above)
Comment (joeldudleyr3) [reply docs/source/example-code/src/main/java/net/corda/docs/FlowCookbookJava.java]: Addressed.
